File: totosuki/ABC/C/193.py
from math import isqrt

N = int(input())
lim = isqrt(N)

log = set()
cnt = 0
a = 2
while a <= isqrt(N): # a**2 <= Nまで
  b = 2
  while a**b <= N:
    if a**b not in log:
      log.add(a**b)
      cnt += 1
    b += 1
  a += 1

print(N - len(log))

File: OCH_Tao/ABC/B/193.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n").slice(1);
let ans = [];
for (const i of input) {
  const [A, P, X] = i.split(" ").map(x => parseInt(x, 10));
  if (X - A > 0) {
    ans.push(P);
  }
}
console.log(ans.length > 0 ? Math.min(...ans) : -1);